src/View.zig

```diff
@@ -0,0 +1,161 @@
+    const screen = try alloc.create(Screen);
+    screen.* = try Screen.init(
+        alloc,
+        .{
+            .cols = config.max_width,
+            .rows = config.max_height,
+            .x_pixel = config.x_pixel,
+            .y_pixel = config.y_pixel,
+        },
+        unicode,
+    );
+    const window = try alloc.create(Window);
+    window.* = .{
+        .x_off = 0,
+        .y_off = 0,
+        .width = config.max_width,
+        .height = config.max_height,
+        .screen = screen,
+    };
+    return .{
+        .alloc = alloc,
+        .screen = screen,
+        .win = window,
+    };
+}
+
+pub fn deinit(self: *View) void {
+    self.alloc.destroy(self.win);
+    self.screen.deinit(self.alloc);
+    self.alloc.destroy(self.screen);
+}
+
+/// Render Config f/ `toWin()`
+pub const RenderConfig = struct {
+    x: usize = 0,
+    y: usize = 0,
+    width: Extent = .fit,
+    height: Extent = .fit,
+
+    pub const Extent = union(enum) {
+        fit,
+        max: usize,
+    };
+};
+/// Render a portion of this View to the provided Window (`win`).
+/// This will return the bounded X (col), Y (row) coordinates based on the rendering.
+pub fn toWin(self: *View, win: *const Window, config: RenderConfig) !struct { usize, usize } {
+    var x = @min(self.screen.width - 1, config.x);
+    var y = @min(self.screen.height - 1, config.y);
+    const width = width: {
+        var width = switch (config.width) {
+            .fit => win.width,
+            .max => |w| @min(win.width, w),
+        };
+        width = @min(width, self.screen.width);
+        break :width @min(width, self.screen.width -| 1 -| x +| win.width);
+    };
+    const height = height: {
+        var height = switch (config.height) {
+            .fit => win.height,
+            .max => |h| @min(win.height, h),
+        };
+        height = @min(height, self.screen.height);
+        break :height @min(height, self.screen.height -| 1 -| y +| win.height);
+    };
+    x = @min(x, self.screen.width -| width);
+    y = @min(y, self.screen.height -| height);
+
+    for (0..height) |row| {
+        for (0..width) |col| {
+            win.writeCell(
+                col,
+                row,
+                self.win.readCell(
+                    @min(self.screen.width -| 1, x +| col),
+                    @min(self.screen.height -| 1, y +| row),
+                ) orelse {
+                    std.log.err(
+                        \\ Position Out of Bounds:
+                        \\ - Pos:  {d}, {d}
+                        \\ - Size: {d}, {d}
+                    ,
+                        .{
+                            col,               row,
+                            self.screen.width, self.screen.height,
+                        },
+                    );
+                    return error.PositionOutOfBounds;
+                },
+            );
+        }
+    }
+    return .{ x, y };
+}
+
+/// Writes a cell to the location in the View
```

Comment:
I think it would make sense to take all of these `win` API wrappers and remove them. You can add a `window` method which returns a root window of the backing screen and the full `window` API is accessible from that.

Reply:
I had two main reasons for this:

1. API simplicity. To me it makes more sense to remove an additional call from users if they're always going to have to make that call to use a View. Otherwise, users will always have to do `view.win().someFn()` instead of just `view.someFn()`. This has the downside of maintaining API parity between Views and Windows, but that leads to the second point.

2. Windows and Views are non 1-for-1. While the View API largely mirrors that of Window, there are a few functions that I don't think View should include. Namely `hasMouse()` and `hideCursor()` (and possibly `scroll()`?) shouldn't be used from View. Those actions should be done from whatever Window the user pushes the View to. That list could also expand in the future if Window sees more changes.